src/__internal__/focus-trap/focus-trap.spec.tsx

```diff
@@ -0,0 +1,779 @@
+      screen.getByLabelText("input").blur();
+      expect(screen.getByLabelText("input")).not.toBeFocused();
+      rerender(
+        <MockComponent autoFocus={false} triggerRefocusFlag>
+          <button type="button">{BUTTON_ONE}</button>
+          <Select
+            id="c0499f86-d5a7-4a72-a0b7-753a2d218c54"
+            label="the dropdown"
+            aria-label="input"
+          >
+            <Option value="1" text="Option 1" />
+          </Select>
+        </MockComponent>
+      );
+      expect(screen.getByLabelText("input")).toBeFocused();
+    });
+
+    it("refocuses the wrapper element when flag is set, if the wrapper has a tabindex", () => {
+      const { rerender } = render(
+        <MockComponent
+          autoFocus={false}
+          triggerRefocusFlag={false}
+          tabIndex={-1}
+        >
+          <button type="button">{BUTTON_ONE}</button>
+          <input type="text" />
+        </MockComponent>
+      );
+      rerender(
+        <MockComponent autoFocus={false} triggerRefocusFlag tabIndex={-1}>
+          <button type="button">{BUTTON_ONE}</button>
+          <input type="text" />
+        </MockComponent>
+      );
+      expect(screen.getByTestId(WRAPPER_ID)).toBeFocused();
+    });
+
+    it("refocuses the container within the trap when flag is set, if the wrapper has no tabindex", () => {
+      const { rerender } = render(
+        <MockComponent autoFocus={false} triggerRefocusFlag={false}>
+          <button type="button">{BUTTON_ONE}</button>
+          <input type="text" />
+        </MockComponent>
+      );
+      rerender(
+        <MockComponent autoFocus={false} triggerRefocusFlag>
+          <button type="button">{BUTTON_ONE}</button>
+          <input type="text" />
+        </MockComponent>
+      );
+      expect(screen.getByTestId(WRAPPER_ID)).toBeFocused();
+    });
+
+    it("refocuses the container if last element that had focus becomes disabled", () => {
+      const { rerender } = render(
+        <MockComponent autoFocus={false} triggerRefocusFlag={false}>
+          <button type="button">{BUTTON_ONE}</button>
+          <input aria-label="input" type="text" id="disable-on-focus" />
+        </MockComponent>
+      );
+      focusElement(screen.getByLabelText("input"));
+      rerender(
+        <MockComponent autoFocus={false} triggerRefocusFlag>
+          <button type="button">{BUTTON_ONE}</button>
+          <input type="text" id="disable-on-focus" />
+        </MockComponent>
+      );
+      expect(screen.getByTestId(WRAPPER_ID)).toBeFocused();
+    });
+  });
+
+  describe("when autoFocus is false", () => {
+    it("should not focus the first focusable element by default", () => {
+      render(
+        <MockComponent autoFocus={false}>
+          <button type="button">{BUTTON_ONE}</button>
+          <input type="text" />
+        </MockComponent>
+      );
+      expect(document.querySelectorAll("body")[0]).toBeFocused();
```

Comment:
```suggestion
      const button = screen.getByRole("button", { name: BUTTON_ONE });
      expect(button).not.toHaveFocus();
```

What do you think of this?

Reply:
I'm happy to add this in, I'll probably leave the other one in to assert that focus wasn't moved etc